components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallback.java

```diff
@@ -185,13 +185,15 @@ public void handleCustomClaims(JWTClaimsSet jwtClaimsSet, OAuthAuthzReqMessageCo
     private Map<String, Object> getResponse(OAuthTokenReqMessageContext requestMsgCtx)
             throws OAuthSystemException {
 
+        // get the user claims cached against the access token if any
         Map<ClaimMapping, String> userAttributes =
-                getUserAttributesFromCache(requestMsgCtx.getProperty(OAuthConstants.ACCESS_TOKEN).toString());
+                getUserAttributesFromCacheUsingToken(requestMsgCtx.getProperty(OAuthConstants.ACCESS_TOKEN).toString());
```

Comment:
Needs a null check similar to 'requestMsgCtx.getProperty(OAuthConstants.AUTHZ_CODE) != null' here as well.